### ISSUE TITLE
Allow Polyfill Imports

### ISSUE THREAD

**Issue by svenanders** [amilajack/eslint-plugin-compat#70]
This is probably hard but I don't think this plugin should warn about usages of APIs that are polyfilled. For instance, say I have a component that uses fetch. With the compat plugin I get:

    error  fetch is not supported in Safari 9.1, Opera Mini all, iOS Safari 9.3, IE Mobile 10, IE 10, Edge 13, Android Browser 4.4, Android UC Browser 11  compat/compat

When I add a polyfyll in this component, for instance by adding _import fetch from ('some-fetch-lib');_ at the top of the file, the plugin should not display the warning. This helps me remember to add a fetch polyfill when needed.

I could add the polyfill to the config, but then I could easily launch the code without importing a fetch polyfill, and the site would be broken for browsers without fetch support.

**Comment by amilajack** [comment]
Ahh I see. Yea that makes sense. That kind of configuration was actually intended for webpack users who could import all their polyfills at bundle time and don't have to do so for every single file.  I'm trying to find ways to use something like flow to trace imports and types for better static analysis. That'll probably be in the next release.

**Comment by ljharb** [comment]
I should also be able to create a function called `fetch` and it shouldn't error - in other words, you always will (and should) need to check if a global var is actually global, and not declared or imported.

**Comment by amilajack** [comment]
> check if a global var is actually global

Can you provide an example of this. Not sure exactly what you mean.

**Comment by ljharb** [comment]
```js
var fetch = function () {};

fetch() // should not error
```

```js
fetch() // should error
```

In other words, when parsing the file, you should be tracking *all* identifiers declared/created/used, and then at the end of the file, looking at all the *potential* problems you accumulated during parsing, and then using the information you gathered to filter out the things that simply coincidentally share the same identifier name as the problematic APIs, and then reporting on what's left.

**Comment by ljharb** [comment]
I believe `eslint-plugin-import` does this in a number of places to figure out unused imports, for example.

**Comment by amilajack** [comment]
Thanks for the clarification. I'm thinking of using something like [flow-inspect](https://github.com/flowtype/flow-inspect) to determine if an API is defined or not. [Here's my reasoning](https://github.com/amilajack/eslint-plugin-compat/wiki/Ideas#type-analysis). Thoughts on this?

**Comment by ljharb** [comment]
I'm not sure why a type system would help specifically - type systems can't check if things are references to the global any differently than linters or parsers can.

**Comment by amilajack** [comment]
> I believe `eslint-plugin-import` does this in a number of places to figure out unused imports, for example.

I couldn't find the `eslint-plugin-import` rule for this. What do you think about using the logic from the `no-unused-vars` rule?

**Comment by amilajack** [comment]
This issue was resolved in https://github.com/amilajack/eslint-plugin-compat/pull/207 🎉
